TaskWebApp/Utils/MsalAppBuilder.cs

```diff
@@ -51,7 +54,16 @@ public static IConfidentialClientApplication BuildConfidentialClientApplication(
                   .WithB2CAuthority(Globals.B2CAuthority)
                   .Build();
 
-            MSALPerUserMemoryTokenCache userTokenCache = new MSALPerUserMemoryTokenCache(clientapp.UserTokenCache, currentUser ?? ClaimsPrincipal.Current);
+            clientapp.AddDistributedTokenCache(services =>
+            {
+                services.AddDistributedMemoryCache();
+                services.Configure<MsalDistributedTokenCacheAdapterOptions>(o =>
+                {
+                    o.Encrypt = true;
+                });
+            });
+
+            new MSALPerUserMemoryTokenCache(clientapp.UserTokenCache, currentUser ?? ClaimsPrincipal.Current);
```

Comment:
you don't need it any longer, @aremo-ms 
AddDistributedTokenCache takes care of both app and user token caches